### PR TITLE
Improve the "must use" lint for `Future`

### LINKED ISSUE #60797: improve the "must use" lint for `Future` to suggest await
The following code [gives a "must use" warning](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=1dce9a2ee23813a18dfcf2b4d6bfa2fe) (which is awesome!), but the warning is not as clear as it could be:

```rust
#![feature(async_await)]
#![allow(dead_code)]

async fn foo() {
    bar();
}

async fn bar() {
}

fn main() { }
```

This results in:

```
warning: unused implementer of `std::future::Future` that must be used
 --> src/main.rs:5:5
  |
5 |     bar();
  |     ^^^^^^
  |
  = note: #[warn(unused_must_use)] on by default
  = note: futures do nothing unless polled
```

it should probably say "futures do nothing unless you await them" or something. Most people won't know what 'polled' is all about, I think.

### PR DESCRIPTION
**Pull request by Schultzer**
Fixes #60797

### CODE REVIEW

File: src/libcore/future/future.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ use crate::task::{Context, Poll};
 /// When using a future, you generally won't call `poll` directly, but instead
 /// `await!` the value.
 #[doc(spotlight)]
-#[must_use = "futures do nothing unless polled"]
+#[must_use = "futures stay idle unless you await them"]
```
Comment (Centril) [inline src/libcore/future/future.rs]:
```suggestion
#[must_use = "futures stay idle unless you `.await` them"]
```
To hint at the surface syntax :)

Comment (Schultzer) [reply src/libcore/future/future.rs]: Thanks :)

File: src/libcore/future/future.rs [hunk 2]
```diff
@@ -23,7 +23,7 @@ use crate::task::{Context, Poll};
 /// When using a future, you generally won't call `poll` directly, but instead
 /// `await!` the value.
 #[doc(spotlight)]
-#[must_use = "futures do nothing unless polled"]
+#[must_use = "futures stay idle unless you `.await` them"]
```
Comment (varkor) [inline src/libcore/future/future.rs]: This is nit-picking, but "stay idle" is indicative of a process to me. "futures are not executed" or "futures do nothing", etc. seems clearer to me. What do you think @Centril?

Comment (cramertj) [reply src/libcore/future/future.rs]: +1

**Comment by cramertj** [comment]
It's probably worth calling out here that `.await` is not the only way to poll a future-- there are many others, like polling directly, passing it to an executor, or using it inside the `join!` or `select!` macros. I think it's helpful to point users towards `await!`, but it might be worth thinking about how to clarify that `await` isn't the *only* thing you can do.

**Comment by Centril** [comment]
@cramertj Open to suggestions on how to point at a broader set, but remember that the message has to be short and sweet.

File: src/libcore/future/future.rs (hunk 2 above)
Comment (Centril) [reply src/libcore/future/future.rs]: I don't feel strongly so let's stick with `do nothing` then.

@Schultzer Can you please adjust `stay idle` to `do nothing` (but keep `or poll`)?

Comment (Schultzer) [reply src/libcore/future/future.rs]: done

**Comment by Centril** [comment]
@bors r+ rollup

There may be some tests that need to be `--bless`ed but I'll sign off optimistically for now.

@cramertj If you think we need to make more changes, feel free to send in another PR or r- if you have a better wording on your mind now.

**Comment by cramertj** [comment]
nah the updated message seems fine to me, thanks :)

**Pull request merged**